### PR TITLE
Remove quoted.matching.Sym

### PR DESCRIPTION
**Pull request by nicolasstucki**
Since quoted patterns do not return open quotes, this abstraction does not have any purpose.
The only useful part of it is extracting the name of a binding, therefore instead of returning a Sym we return the name as a String directly.

### CODE REVIEW

File: tests/pos/quotedPatterns.scala [hunk 1]
```diff
@@ -14,22 +14,22 @@ object Test {
     case '{ 1 + ($y: Int)} => y
     case '{ val a = 1 + ($y: Int); 3 } => y
     case '{ val $y: Int = $z; println(`$y`); 1 } =>
-      val a: quoted.matching.Sym[Int] = y
+      val a: String = y
```
Comment (liufengyun) [inline tests/pos/quotedPatterns.scala]: If `y` is String, the semantics for the reference to `y` in `println($y)` becomes problematic.

Comment (nicolasstucki) [reply tests/pos/quotedPatterns.scala]: Why? Could you elaborate

Comment (liufengyun) [reply tests/pos/quotedPatterns.scala]: After discussion, now I see the `$y` in `val $y = e` has double meaning.

**Comment by nicolasstucki** [comment]
Rebased

File: tests/pos/quotedPatterns.scala (hunk 1 above)
Comment (nicolasstucki) [reply tests/pos/quotedPatterns.scala]: In general, a `$` can be in an expression position (`Expr[T]`), type position (`Type[T]`) or definition position (`String`)

**Pull request merged**